x-pack/plugins/ml/public/application/model_management/deployment_setup.tsx

```diff
@@ -63,300 +129,496 @@ export const DeploymentSetup: FC<DeploymentSetupProps> = ({
   errors,
   isUpdate,
   deploymentsParams,
+  disableAdaptiveResourcesControl,
+  deploymentParamsMapper,
+  cloudInfo,
+  showNodeInfo,
 }) => {
-  const {
-    total_ml_processors: totalMlProcessors,
-    max_single_ml_node_processors: maxSingleMlNodeProcessors,
-  } = getNewJobLimits();
-
-  const numOfAllocation = config.numOfAllocations;
-  const threadsPerAllocations = config.threadsPerAllocations;
+  const deploymentIdUpdated = useRef(false);
 
   const defaultDeploymentId = useMemo(() => {
     return config.deploymentId;
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, []);
 
-  const threadsPerAllocationsOptions = useMemo(
-    () =>
-      new Array(THREADS_MAX_EXPONENT)
-        .fill(null)
-        .map((v, i) => Math.pow(2, i))
-        .filter(maxSingleMlNodeProcessors ? (v) => v <= maxSingleMlNodeProcessors : (v) => true)
-        .map((value) => {
-          const id = value.toString();
-
-          return {
-            id,
-            label: id,
-            value,
-            'data-test-subj': `mlModelsStartDeploymentModalThreadsPerAllocation_${id}`,
-          };
-        }),
-    [maxSingleMlNodeProcessors]
-  );
-
-  const disableThreadingControls = config.priority === 'low';
-
-  return (
-    <EuiForm component={'form'} id={'startDeploymentForm'}>
-      <EuiDescribedFormGroup
-        titleSize={'xxs'}
-        title={
-          <h3>
+  const customTicks = Object.values(vCpuLevelMap).map((v) => {
```

Comment:
nit, `customTicks`, `customColorsLevels` and `optimizedOptions` could be wrapped in useMemos

Reply:
updated in a7b1feaa90a7658af57929ac653741bccc3bfb3a